pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3063,6 +3046,27 @@ public void onUpdate(TopicPolicies policies) {
         });
     }
 
+    private CompletableFuture<Void> checkReplicator() {
```

Comment:
There is already a `checkReplication` in this class. It seems contains most of the logic here.

Reply:
Yes, thanks.